### PR TITLE
fix Issue 15353 - ignore freeing during finalization

### PR DESCRIPTION
**Pull request by MartinNowak**

### CODE REVIEW

File: changelog.dd [hunk 1]
```diff
@@ -42,6 +43,12 @@ Macros:
     COREMODREF = <a href="phobos/core_$1.html">$2</a>
     XREF = <a href="phobos/std_$1.html#$2">$2</a>
     CXREF = <a href="phobos/core_$1.html#$2">$2</a>
+    OXREF = <a href="phobos/object.html#$2">$2</a>
+    NXREF = <a href="phobos/std_$1.html#.$2">$2</a>
```
Comment (andralex) [inline changelog.dd]: Are these other added macros used?

**Comment by andralex** [comment]
Nice, thanks. I'll pull this although there's a question about the extra macros.

**Comment by andralex** [comment]
Auto-merge toggled on

File: changelog.dd (hunk 1 above)
Comment (MartinNowak) [inline changelog.dd]: I just keep them in sync w/ what's available in dlang.org, so people don't fall back to plain html.

File: src/gc/gc.d [hunk 2]
```diff
@@ -3250,6 +3250,23 @@ unittest // bugzilla 14467
...
+    static struct Foo
+    {
+        ~this()
+        {
+            GC.free(buf); // ignored in finalizer
+        }
+
+        void* buf;
+    }
+    new Foo(GC.malloc(10));
```
Comment (CyberShadow) [inline src/gc/gc.d]: This should probably be in a loop, to make sure there is no left-over pointer in the registers or stack.

**Pull request merged**